src/iuse.cpp

```diff
@@ -6439,20 +6539,788 @@ struct npc_photo_def : public JsonDeserializer, public JsonSerializer {
+                                             string_format( _( " from %s" ), veh_name );
+                if( ret_obj.vehicles.find( veh_name ) != ret_obj.vehicles.end() ) {
+                    ret_obj.vehicles.erase( veh_name );
+                }
+            }
+            local_vehicles_recorded.insert( veh_hash );
+        } else if( !item.is_null() ) {
+            item_name = trap_name + item_name + field_desc;
+            if( point == point_around_figure && create_figure_desc ) {
+                description_terrain_on_figure = ter_desc + _( " with " ) + item_name;
+            } else {
+                ret_obj.items[ item_name ] ++;
+            }
+        } else if( !unusual_ter_desc.empty() ) {
+            unusual_ter_desc = trap_name + unusual_ter_desc + field_desc;
+            if( point == point_around_figure && create_figure_desc ) {
+                description_furniture_on_figure = unusual_ter_desc;
+            } else {
+                ret_obj.furniture[ unusual_ter_desc ] ++;
+            }
+        } else if( !ter_desc.empty() && ( !field_desc.empty() || !trap_name.empty() ) ) {
+            ter_desc = trap_name + ter_desc + field_desc;
+            if( point == point_around_figure && create_figure_desc ) {
+                description_terrain_on_figure = ter_desc;
+            } else {
+                ret_obj.terrain[ ter_desc ] ++;
+            }
+        } else {
+            ter_desc = trap_name + ter_desc + field_desc;
+            if( point == point_around_figure && create_figure_desc ) {
+                description_terrain_on_figure = ter_desc;
+            }
+        }
+        ignored_points.insert( point_around_figure );
+    }
+
+    if( create_figure_desc ) {
+        std::vector<std::string> objects_combined_desc;
+        std::unordered_map<std::string, int> vecs_to_retrieve[4] = {
+            ret_obj.furniture, ret_obj.vehicles, ret_obj.items, ret_obj.terrain
+        };
+
+        for( int i = 0; i < 4; i++ ) {
+            for( const auto &p : vecs_to_retrieve[ i ] ) {
+                objects_combined_desc.push_back( format_object_pair( p ) );
+            }
+        }
+
+        const char *transl_str = pgettext( "someone stands/sits *on* something", " on %s." );
+        if( !description_part_on_figure.empty() ) {
+            ret_obj.figure_text = string_format( transl_str, description_part_on_figure );
+        } else {
+            if( !description_furniture_on_figure.empty() ) {
+                ret_obj.figure_text = string_format( transl_str, description_furniture_on_figure );
+            } else {
+                ret_obj.figure_text = string_format( transl_str, description_terrain_on_figure );
+            }
+        }
+        if( !objects_combined_desc.empty() ) {
+            // store objects to description_figures_status
+            std::string objects_text = enumerate_as_string( objects_combined_desc );
```

Comment:
We could give these objects articles to make it read a little better
```suggestion
            std::string objects_text = enumerate_as_string( objects_combined_desc.begin(), objects_combined_desc.end(), []( const std::string &obj){
                return "a " + obj; 
            } );
```